### ISSUE TITLE
i18n: po not compiled

### ISSUE THREAD

**Issue by dschwoerer** [git-cola/git-cola#1252]
`make install` installs uncompiled `.po` files.

I tried the resulting installation, and git-cola UI was not translated.

**Comment by davvid** [comment]
The whole PEP-517/518 kerfuffle made it so that the custom distutils commands we were using were no longer maintainable. These commands were the ones that compiled the po files.

I then figured that using package_data (instead of the unsupported "data_files" setuptools feature) alongside a copy of the [polib](https://pypi.org/project/polib/) module would be a way to handle it. We would just read the po files directly.

I have tested swapping the language around using `export LANG=ja_JP.UTF_8` but I haven't tested beyond that (and the unit tests that verify that the i18n machinery is functional).

We were previously using the gettext module to detect the locale and setup the translations. Now that we have to use polib (and do not have precompiled .mo files) our logic for detecting the locale may need some enhancements. I suspect our detection may be insufficient.

Thanks for the heads-up. What method are you using to test using a translated language? This seems worth fixing before tagging the next patch release.
